backEnd/index.js

```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
-      name: task.name,
-      fileUpload: null,
-      dueDate: dueDate,
-      isApproved: task.isApproved,
-      isSubmitted: false,
-    })
-  });
+  try {
+    userValidator.validateUser(toCreate);
 
-  let accountCreationDate = new Date();
-  let notificationDate = accountCreationDate.toLocaleDateString("en-CA", {
-    timeZone: "America/Edmonton"
-  });
-
-  //TODO: Add new user to firebase. Currently adds to server RAM
-  applicants.push({
-    id: newUser.id,
-    name: newUser.name,
-    email: newUser.email,
-    password: newUser.password, // TODO: Salt and hash the password or make this work with firebase authentication
-    notifications: [
+    toCreate.tasks =  taskFactory.getDefaultTasks();
+    // password: newUser.password, // TODO: Salt and hash the password or make this work with firebase authentication
+    toCreate.notifications = [
       {
         message: "Congratulations on making your account!",
-        date: notificationDate
+        date: new Date().toLocaleDateString("en-CA", { timeZone: "America/Edmonton" })
       }
-    ],
-    isAdmin: false,
-    isCommunityMentor: false,
-    requiresHomeAssessment: false,
-    tasks
-  })
-
-  console.log("New user created: ")
-  console.log(Object.entries(applicants[applicants.length-1]))
-
-  console.log("Tasks: ")
-  tasks.forEach(task => console.log(Object.entries(task)));
-
-
-  res.json(newUser.id + " was created")
+    ];
+    toCreate.isAdmin = false;
+    toCreate.isCommunityMentor = false;
+    toCreate.requiresHomeAssessment =  false;
+      
+    await userRepository.createUser(toCreate); 
+    res.json(toCreate.id)
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
+    res.json(error);
+  }
 });
 
 // Delete user by id
 // receives a user ID to delete from client
-app.delete("/api/users/:id", (req, res) => {
+app.delete("/api/users/:id", async (req,res) => {
   let id = req.params.id;
   console.log("Delete User by ID: /users/:" + id);
-  //todo: find user in firebase and delete
-  res.json(id + " was deleted");
+  try {
+    userValidator.validateDelete(id);
+    await userRepository.deleteUser(id);
+    res.json(id + " was deleted");
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
+    res.json(error);
+  }
 });
 
-// Update user task - this can be called by the admin OR user
+// Update user - this can be called by the admin OR user
 // receives a json from the client
-app.put("/api/users", (req, res) => {
-  let updatedUser = req.body.user;
-  console.log("Updating user");
-  console.log(updatedUser);
-  //todo: Update user in Firebase
-  res.json(updatedUser);
+app.put("/api/users", async (req,res) => {
+  let toUpdate = req.body;
+  try {
+    userValidator.validateUser(toUpdate);
+
+    console.log("Updating user");
+    console.log(toUpdate);
+    await userRepository.updateUser(toUpdate);
+    res.json(toUpdate);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```

Comment:
404 if the requested client doesn't exist. 500 if the server can't update the requested client. Request validation should be done on the client side so there shouldn't be a malformed request (a 400 error)